### ISSUE TITLE
Incorrect Highlighting For Multiple Line Type Signatures

### ISSUE THREAD

**Issue by mizunashi-mana** [JustusAdam/language-haskell#1]
Thanks for awesome plugin!

I wrote haskell codes such as:

``` haskell
dbPageWrapper :: (DashBoardPage p, Monad m) =>
  T.Text -> p -> HtmlT m a -> HtmlT m b -> HtmlT m b -- disable type highlighting
dbPageWrapper pref page headerC mainC = do
  ...
```

Then, I noticed missing syntax.

I use GHC 7.10.3.  Probably, because building is passed, this is legal syntax.

**Comment by JustusAdam** [comment]
Yes, I know what you mean. I believe this is a shortcoming of the Haskell [texmate bundle](/textmate/haskell.tmbundle). I will open an issue there, perhaps we can fix this.

**Comment by mizunashi-mana** [comment]
@JustusAdam I see. Thanks for your excellent understanding!

**Comment by JustusAdam** [comment]
I took a closer look at the regexes myself and it almost seems to me to be not an issue of the regex itself, but it seems that the matching algorithm is designed to match line-by-line and the regex is written such that it matches the entire signature at once.

I'll investigate this further and perhaps change the matchings to match partial signatures, which naturally should then allow multi line type signatures to be matched correctly, hopefully.

**Comment by JustusAdam** [comment]
Resolved as of 271fd41e4f3ca3afafea74049d5d5a1c91bf816d

**Comment by mizunashi-mana** [comment]
@JustusAdam Thanks for your great works!
